.eslintrc.json

```diff
@@ -0,0 +1,15 @@
+{
+  "extends": "plugin:@tinymce/standard",
+  "parserOptions": {
+    "sourceType": "module",
+    "project": [
+      "./tsconfig.json"
+    ]
+  },
+  "rules": {
+    "no-underscore-dangle": "off",
+    "@typescript-eslint/explicit-member-accessibility": "off",
+    "@tinymce/prefer-fun": "off",
+    "@typescript-eslint/member-ordering": "off"
+  }
+}
```

Comment:
This will lint the js files incorrectly because our linting rules only work for typescript. You can do both (or just ignore the js files) if you put the linting rules in the overrides section. See connectfour for an example.
```json
{
  "overrides": [
    {
      "files": [
        "**/*.ts"
      ],
      "extends": "plugin:@tinymce/standard",
      "parserOptions": {
        "project": "tsconfig.json",
        "sourceType": "module"
      },
      "rules": {
        "quote-props": ["error", "always"],
        "no-shadow": "off",
        "@typescript-eslint/no-unsafe-assignment": "off",
        "@typescript-eslint/no-unsafe-call": "off",
        "@typescript-eslint/no-unsafe-member-access": "off",
        "@typescript-eslint/no-unsafe-return": "off",
        "@typescript-eslint/explicit-module-boundary-types": "off",
        "@typescript-eslint/restrict-plus-operands": "off"
      }
    },
    {
      "files": [
        "rollup.config.js",
        "bin/**/*.js",
        "src/**/*.js"
      ],
      "excludedFiles": ["src/main/js/footer.js", "src/main/js/header.js", "src/main/render/prism.js"],
      "env": {
        "es6": true,
        "node": true,
        "browser": true
      },
      "extends": "eslint:recommended",
      "parser": "espree",
      "parserOptions": {
        "ecmaVersion": 2020,
        "sourceType": "module"
      },
      "rules": {
        "indent": [ "error", 2, { "SwitchCase": 1 } ],
        "no-shadow": "error",
        "no-unused-vars": [ "error", { "argsIgnorePattern": "^_" } ],
        "object-curly-spacing": [ "error", "always", { "arraysInObjects": false, "objectsInObjects": false } ],
        "quotes": [ "error", "single" ],
        "semi": "error"
      }
    }
  ]
}
```
In the future we should probably make our eslint rules also export a js safe version.

Reply:
The package.json has a glob that specifies only ts files. The only js files are rollup.config.js and src/demo/server.js. The latter could probably be converted to TS pretty easily.

The other way you can have different rules is to have separate eslint config files in different folders. 

I think it's probably fine to just ignore the JS for this project.

Reply:
Yes the `yarn lint` command is fine but if you open any of those js files in vscode then you'll get linting errors. I guess you've been using intellij so you haven't noticed.

Reply:
Ah. Righto, I'll fix that up.

Reply:
Hmmm, I stand corrected. I checked it out and I don't get linting errors like I expected.

Reply:
Weird.

Reply:
I ported the Server to TypeScript. Can you please review that?

Reply:
Looks fine to me.